localstack/services/events/target.py

```diff
@@ -33,6 +49,38 @@ def transform_event_with_target_input_path(
     return formatted_event
 
 
+def get_template_replacements(
+    input_transformer: InputTransformer, event: FormattedEvent
+) -> dict[str, Any]:
+    """Extracts values from the event using the input paths map keys and places them in the input template dict."""
+    template_replacements = {}
+    transformer_path_map = input_transformer.get("InputPathsMap", {})
+    for placeholder, transformer_path in transformer_path_map.items():
+        if placeholder in PREDEFINED_PLACEHOLDERS:
+            continue
+        value = extract_jsonpath(event, transformer_path)
+        if not value:
+            value = ""  # default value is empty string
+        template_replacements[placeholder] = value
+    return template_replacements
+
+
+def replace_template_placeholders(
+    template: str, replacements: dict[str, Any], is_json: bool
+) -> TransformedEvent:
+    """Replace placeholders defined by <key> in the template with the values from the replacements dict.
+    Can handle single template string or template dict."""
+
+    def replace_placeholder(match):
+        key = match.group(1)
+        value = replacements.get(key, match.group(0))  # handle non defined placeholders
+        return json.dumps(value) if is_json else value
```

Comment:
How are non-string values such as the discussed integer scenario work now?

Reply:
I don't deal with this one edge case for for the predefined placeholder <aws.events.events> where the event body contains "time" which is returned as an int, not a string. I figure that this is such an edge case that we can deal with it once we have the provider at parity.

Reply:
Fair point 👍 
Let's keep track of it somewhere for the future.